### PR TITLE
task4601 report post

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4601

### CODE REVIEW

File: app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -263,6 +263,39 @@ public function postBestAnswer(Request $request)
...
+    public function postReport(Request $request)
+    {
+        $input = $request->all();
+
+        if ($request->ajax() && !empty($input['post_id']) && Auth::check()) {
+            $input['user_id'] = Auth::id();
+            $input['comment'] = config('constants.REPORT_MESSAGES.' . $input['report'], null);
+            $input['comment'] = $input['comment'] . '|||' .$input['note'];
+
+            $validator = \Validator::make($input,
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 275</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -263,6 +263,39 @@ public function postBestAnswer(Request $request)
...
+            [
+                'note' => 'max:200',
+                'report' => 'required|numeric',
+            ],
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 279</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php [hunk 3]
```diff
@@ -263,6 +263,39 @@ public function postBestAnswer(Request $request)
...
+            [
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 280</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php [hunk 4]
```diff
@@ -263,6 +263,39 @@ public function postBestAnswer(Request $request)
...
+                'max' => __('validation.max.string', ['attribute' => __('page.post.note'), 'max' => 200]),
+                'report.required' => __('validation.required', ['attribute' => __('page.post.report')]),
+            ]);
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 283</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Http/Controllers/PostController.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 275</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/PostController.php [hunk 5]
```diff
@@ -263,6 +263,39 @@ public function postBestAnswer(Request $request)
...
+    {
+        $input = $request->all();
+
+        if ($request->ajax() && !empty($input['post_id']) && Auth::check()) {
+            $input['user_id'] = Auth::id();
+            $input['comment'] = config('constants.REPORT_MESSAGES.' . $input['report'], null);
+            $input['comment'] = $input['comment'] . '|||' .$input['note'];
+
+            $validator = \Validator::make($input,
+            [
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 276</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php (hunk 2 above)
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 279</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php (hunk 3 above)
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 280</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php (hunk 4 above)
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 283</br>Closing parenthesis of a multi-line function call must be on a line by itself

**Comment by laam55** [comment]
ready

File: resources/js/post-detail.js [hunk 6]
```diff
@@ -35,7 +35,57 @@ function ValidURL(str) {
     }
 }
 
+jQuery.fn.center = function () {
+    this.css("position", "absolute");
+    this.css("top", ($(window).height() - this.height()) / 2 + $(window).scrollTop() - 100 + "px");
+    this.css("left", ($(window).width() - this.width()) / 2 + $(window).scrollLeft() + "px");
```
Comment (BunnyPi04) [inline resources/js/post-detail.js]: sử dụng nháy đơn

File: app/Report.php [hunk 7]
```diff
@@ -26,4 +27,15 @@ public function user()
     {
         return $this->belongsTo('App\User');
     }
+
+    public function saveReport($input)
+    {
+        if (!empty($input['comment'])) {
+            $report = new Report($input);
+
+            return $report->save();
```
Comment (BunnyPi04) [inline app/Report.php]: sao ko dùng Report::create($input) luôn cho nhanh?

File: resources/views/home/layout/modal.blade.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        {!! Form::open(['id' => 'reportForm', 'class' => 'searchbar js-searchbar', 'route' => 'home.post.postReport']) !!}
+        <div class="anon-vote-blurb">
+            <p><span class="errors">*</span> {{ __('page.report.reason') }}</p>
+
+            @foreach (config('constants.REPORT_MESSAGES') as $key => $report)
+            {{ Form::radio('report', $key, true, ['id' => 'labelFor' . $key]) }} {!! Form::label('labelFor' . $key, $report) !!}
+            <br>
+            @endforeach
+            <span class="errors" id="reportError"></span>
+                {!! Form::textarea('note', null, ['id' => 'note', 'rows' => '8', 'placeholder' => 'typing reason for reporting this content...']) !!}
```
Comment (BunnyPi04) [inline resources/views/home/layout/modal.blade.php]: i18n

File: app/Report.php (hunk 7 above)
Comment (laam55) [reply app/Report.php]: uh. hihi@@

File: app/Http/Controllers/PostController.php [hunk 9]
```diff
@@ -263,6 +263,41 @@ public function postBestAnswer(Request $request)
...
     }
 
+    public function postReport(Request $request)
+    {
+        $input = $request->all();
+
+        if ($request->ajax() && !empty($input['post_id']) && Auth::check()) {
+            $input['user_id'] = Auth::id();
+            $input['comment'] = config('constants.REPORT_MESSAGES.' . $input['report'], null);
+            $input['comment'] = $input['comment'] . '|||' .$input['note'];
```
Comment (BunnyPi04) [inline app/Http/Controllers/PostController.php]: convention

File: resources/views/home/layout/modal.blade.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
+<div id="report-modal" class="anon-vote-popup vote-popup popup custom-popup">
+    <div class="popup-close">
+        <a title="close this popup">×</a>
```
Comment (BunnyPi04) [inline resources/views/home/layout/modal.blade.php:3]: Suggest: nên dùng icon close ấy, cho đẹp
của font awesome có sẵn

File: resources/views/home/master.blade.php [hunk 11]
```diff
@@ -33,7 +33,6 @@
     <script src="{{ asset('js/manifest.js') }}"></script>
     <script src="{{ asset('js/vendor.js') }}"></script>
     <script src="{{ asset('js/app.js') }}"></script>
-    <script src="{{ asset('js/custom.js') }}"></script>
     <script>
         $(document).ready(function() {
             $('.nuxt-progress').animate({ width: "100%" }, 1000, function() {
```
Comment (BunnyPi04) [inline resources/views/home/master.blade.php]: sử dụng nháy đơn

**Pull request merged**